webcomponents/vectortile-view-component/src/app/feature-view/feature-view.component.ts

```diff
@@ -0,0 +1,285 @@
+import { Extent, getCenter, getTopLeft } from 'ol/extent'
+import { Geometry } from 'ol/geom'
+import { fromExtent } from 'ol/geom/Polygon'
+import { DragBox } from 'ol/interaction'
+import { Group, Tile, Vector as VectorLayer } from 'ol/layer'
+import TileLayer from 'ol/layer/Tile'
+import { Projection, ProjectionLike } from 'ol/proj'
+import { OSM, Vector as VectorSource, WMTS as WMTSSource } from 'ol/source'
+import { Circle, Fill, Stroke, Style } from 'ol/style'
+import WMTSTileGrid from 'ol/tilegrid/WMTS'
+import { take } from 'rxjs/operators'
+import { NgChanges } from '../app.component'
+import { DataUrl, FeatureServiceService } from '../feature-service.service'
+import { featureCollectionGeoJSON } from '../openapi/models/featureCollectionGeoJSON'
+
+export function exhaustiveGuard(_value: never): never {
+  throw new Error(`ERROR! Reached forbidden guard function with unexpected value: ${JSON.stringify(_value)}`)
+}
+
+@Component({
+  selector: 'app-feature-view',
+  templateUrl: './feature-view.component.html',
+  styleUrls: ['./feature-view.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+})
+export class FeatureViewComponent implements OnChanges {
+  @Input() itemsUrl!: string
+  @Input() projection: ProjectionLike = 'EPSG:3857'
+  @Input() backgroundMap: 'BRT' | 'OSM' = 'OSM'
+  @Output() box = new EventEmitter<string>()
+  @Output() activeFeature = new EventEmitter<FeatureLike>()
+  mapHeight = 400
+  mapWidth = 600
+  map: OLMap = this.getMap()
+  featureCollectionGeoJSON!: featureCollectionGeoJSON
+  features: Feature<Geometry>[] = []
+  boxLayer!: VectorLayer<VectorSource<Geometry>>
+
+  constructor(
+    private el: ElementRef,
+    private featureService: FeatureServiceService
+  ) {}
+
+  private getMap(): OLMap {
+    return new OLMap({
+      view: new View({
+        projection: this.projection,
+      }),
+    })
+  }
+
+  private init() {
+    this.mapWidth = this.el.nativeElement.offsetWidth * 0.99
+    this.mapHeight = this.mapWidth * 0.75 // height = 0.75 * width creates 4:3 aspect ratio
+    const mapdiv: HTMLElement = this.el.nativeElement.querySelector('#featuremap')
+    this.map.setTarget(mapdiv)
+    const aurl: DataUrl = { url: this.itemsUrl, projection: this.projection }
+    this.featureService
+      .getFeatures(aurl)
+      .pipe(take(1))
+      .subscribe(data => {
+        this.map.setLayerGroup(new Group())
+        this.features = data
+        this.loadfeatures(this.features)
+        this.loadbackground()
+        this.adddragbox()
+        this.addFeatureEmit()
+      })
+  }
+
+  ngOnChanges(changes: NgChanges<FeatureViewComponent>) {
+    if (
+      changes.itemsUrl?.previousValue !== changes.itemsUrl?.currentValue ||
+      changes.projection.previousValue !== changes.projection.currentValue
+    ) {
+      if (changes.itemsUrl?.currentValue) {
+        this.init()
+      }
+    }
+  }
+
+  loadbackground() {
```

Comment:
Stel voor om overal camelCase te gebruiken. Dus `loadBackground` en `getStyle` etc. Meerdere methods zijn nu in lowercase.

Reply:
pas ik aan